core/src/loader.rs

```diff
@@ -1492,7 +1493,123 @@ impl<'gc> Loader<'gc> {
                             &mut activation.context,
                             "complete",
                         );
-                        Avm2::dispatch_event(uc, complete_evt, sound_object);
+                        Avm2::dispatch_event(&mut activation.context, complete_evt, sound_object);
+
+                        let mut call_id3_event = false;
+                        let id3 = activation
+                            .avm2()
+                            .classes()
+                            .id3info
+                            .construct(&mut activation, &[])
+                            .expect("failed to construct ID3Info object");
+                        if let Avm2Object::ScriptObject(id3) = id3 {
+                            let tag = Tag::read_from(response.body.as_slice());
+                            if let Ok(tag) = tag {
+                                call_id3_event = true;
+                                let _ = id3.set_property_local(
```

Comment:
Can you use `set_public_property` here?

Reply:
Is the VM_INTERNAL api version fine here?